administrator/components/com_media/Controller/ApiController.php

```diff
@@ -56,24 +56,20 @@ public function execute($task)
 				throw new \InvalidArgumentException(\JText::_('JINVALID_TOKEN'), 403);
 			}
 
-			$doTask = strtolower($method . '_' . $task);
+			$doTask = strtolower($method) . ucfirst($task);
 
 			// Record the actual task being fired
 			$this->doTask = $doTask;
 
-			if (in_array($this->doTask, $this->taskMap))
+			if (!in_array($this->doTask, $this->taskMap))
 			{
-				return $this->$doTask();
+				throw new \Exception(\JText::sprintf('JLIB_APPLICATION_ERROR_TASK_NOT_FOUND', $task), 404);
```

Comment:
i would suggest status code 405 instead of 404 to be more explicit.

Reply:
we can probably directly return the response instead of throwing and catching exception in the same method.